src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/HtmlMarkupParser.cs

```diff
@@ -267,7 +609,69 @@ private MarkupStartTagSyntax ParseStartTag(out string tagName, out MarkupTagMode
                     // This is a self closing tag.
                     tagMode = MarkupTagMode.SelfClosing;
                 }
-                TryAccept(SyntaxKind.CloseAngle);
+
+                if (mode == ParseMode.MarkupInCodeBlock)
+                {
+                    if (EndOfFile || !At(SyntaxKind.CloseAngle))
+                    {
+                        // Unfinished tag
+                        complete = false;
+                        Context.ErrorSink.OnError(
+                            RazorDiagnosticFactory.CreateParsing_UnfinishedTag(
+                                new SourceSpan(
+                                    tagName == null ? tagStartLocation : SourceLocationTracker.Advance(tagStartLocation, "<"),
+                                    Math.Max(tagName?.Length ?? 0, 1)),
+                                tagName ?? string.Empty));
+                    }
+                    else
+                    {
+                        complete = TryAccept(SyntaxKind.CloseAngle);
+
+                        // Completed tags in code blocks have no accepted characters.
+                        SpanContext.EditHandler.AcceptedCharacters = AcceptedCharactersInternal.None;
+
+                        if (tagMode != MarkupTagMode.SelfClosing && ParserHelpers.VoidElements.Contains(tagName))
+                        {
+                            // This is a void element.
+                            // Technically, void elements like "meta" are not allowed to have end tags. Just in case they do,
+                            // we need to look ahead at the next set of tokens.
```

Comment:
Even though we decided to treat void tags like any other tags, I needed to do this so we don't error for missing void end tags